File: demo-java/src/main/java/com/playwire/demo_java/ads/view/nativead/NativeView.java
package com.playwire.demo_java.ads.view.nativead;

import android.content.Context;
import android.util.AttributeSet;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.annotation.Nullable;

import com.intergi.playwiresdk.ads.view.nativead.PWNativeViewContent;
import com.playwire.demo_java.R;

public class NativeView extends LinearLayout {

    TextView headlineTextView;
    TextView bodyTextView;
    ImageView appIconImageView;
    TextView storeTextView;
    TextView priceTextView;
    TextView advertiserTextView;
    Button actionButton;
    FrameLayout mediaViewHolder;

    public NativeView(Context context) {
        super(context);
    }

    public NativeView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public NativeView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public NativeView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();

        appIconImageView = findViewById(R.id.ad_app_icon);
        headlineTextView = findViewById(R.id.ad_headline);
        bodyTextView = findViewById(R.id.ad_body);
        actionButton = findViewById(R.id.ad_call_to_action);
        storeTextView = findViewById(R.id.ad_store);
        priceTextView = findViewById(R.id.ad_price);
        advertiserTextView = findViewById(R.id.ad_advertiser);
        mediaViewHolder = findViewById(R.id.ad_media);
    }
    
    void configure(PWNativeViewContent adContent) {
        /// Configure views with ad content.
        headlineTextView.setVisibility(adContent.getHeadline() != null ? View.VISIBLE : View.GONE);
        headlineTextView.setText(adContent.getHeadline());

        /// Hide view in case ad content doesn't contain required information.
        bodyTextView.setVisibility(adContent.getBody() != null ? View.VISIBLE : View.GONE);
        bodyTextView.setText(adContent.getBody());

        actionButton.setVisibility(adContent.getCallToAction() != null ? View.VISIBLE : View.GONE);
        actionButton.setText(adContent.getCallToAction());

        priceTextView.setVisibility(adContent.getPrice() != null ? View.VISIBLE : View.GONE);
        priceTextView.setText(adContent.getPrice());

        storeTextView.setVisibility(adContent.getStore() != null ? View.VISIBLE : View.GONE);
        storeTextView.setText(adContent.getStore());

        advertiserTextView.setVisibility(adContent.getAdvertiser() != null ? View.VISIBLE : View.GONE);
        advertiserTextView.setText(adContent.getAdvertiser());

        appIconImageView.setVisibility(adContent.getIcon() != null ? View.VISIBLE : View.GONE);
        appIconImageView.setImageDrawable(adContent.getIcon());

        mediaViewHolder.setVisibility(adContent.getMediaView() != null ? View.VISIBLE : View.GONE);
        if (adContent.getMediaView() != null) {
            LayoutParams params = new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT);
            params.gravity = Gravity.CENTER_HORIZONTAL;
            params.weight = 1.0f;
            adContent.getMediaView().setLayoutParams(params);
            mediaViewHolder.addView(adContent.getMediaView());
        }
    }
}

File: demo-kotlin/src/main/java/com/playwire/demo_kotlin/ads/view/nativead/NativeView.kt
package com.playwire.demo_kotlin.ads.view.nativead

import android.content.Context
import android.util.AttributeSet
import android.view.Gravity
import android.view.View
import android.widget.*
import com.intergi.playwiresdk.ads.view.nativead.PWNativeViewContent
import com.playwire.demo_kotlin.R

class NativeView: LinearLayout {
    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle)
    constructor(context: Context, attrs: AttributeSet) : this(context, attrs, 0)

    private lateinit var headlineTextView: TextView
    private lateinit var bodyTextView: TextView
    internal lateinit var actionButton: Button
    private lateinit var appIconImageView: ImageView
    private lateinit var storeTextView: TextView
    private lateinit var priceTextView: TextView
    private lateinit var advertiserTextView: TextView
    private lateinit var mediaViewHolder: FrameLayout

    override fun onFinishInflate() {
        super.onFinishInflate()

        appIconImageView = findViewById(R.id.ad_app_icon)
        headlineTextView = findViewById(R.id.ad_headline)
        bodyTextView = findViewById(R.id.ad_body)
        actionButton = findViewById(R.id.ad_call_to_action)
        storeTextView = findViewById(R.id.ad_store)
        priceTextView = findViewById(R.id.ad_price)
        advertiserTextView = findViewById(R.id.ad_advertiser)
        mediaViewHolder = findViewById(R.id.ad_media)
    }

    internal fun configure(adContent: PWNativeViewContent) {
        // Configure views with ad content.
        headlineTextView.visibility = visibleIf { adContent.headline != null }
        headlineTextView.text = adContent.headline

        // Hide view in case ad content doesn't contain required information.
        bodyTextView.visibility = visibleIf { adContent.body != null }
        bodyTextView.text = adContent.body

        actionButton.visibility = visibleIf { adContent.callToAction != null }
        actionButton.text = adContent.callToAction

        priceTextView.visibility = visibleIf { adContent.price != null }
        priceTextView.text = adContent.price

        storeTextView.visibility = visibleIf { adContent.store != null }
        storeTextView.text = adContent.store

        advertiserTextView.visibility = visibleIf { adContent.advertiser != null }
        advertiserTextView.text = adContent.advertiser

        appIconImageView.visibility = visibleIf { adContent.icon != null }
        appIconImageView.setImageDrawable(adContent.icon)

        mediaViewHolder.visibility = visibleIf { adContent.mediaView != null }
        if (adContent.mediaView != null) {
            val params = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT)
            params.gravity = Gravity.CENTER_HORIZONTAL
            params.weight = 1.0f
            adContent.mediaView?.layoutParams = params
            mediaViewHolder.addView(adContent.mediaView)
        }
    }

    private fun visibleIf(isVisible: () -> Boolean): Int {
        return if(isVisible()) View.VISIBLE else View.INVISIBLE
    }
}